dev/core/src/com/google/gwt/dev/jjs/impl/GwtAstBuilder.java

```diff
@@ -2752,6 +2764,12 @@ protected void pushBinaryOp(BinaryExpression x, JBinaryOperator op) {
     }
 
     protected boolean visit(TypeDeclaration x) {
+      if (x.isRecord()) {
+        InternalCompilerException excption =
+            new InternalCompilerException("Records not yet supported");
+        excption.addNode(new JClassType(makeSourceInfo(x), null, false, false));
```

Comment:
```suggestion
        InternalCompilerException exception =
            new InternalCompilerException("Records not yet supported");
        exception.addNode(new JClassType(makeSourceInfo(x), intern(x.name), false, false));
```
Prevents a NPE later in throwing this exception (and fixes a typo).